packages/web-worker/src/pure.ts

```diff
@@ -133,11 +133,11 @@ export function defineWebWorker() {
 
       runner.executeFile(fsPath)
         .then(() => {
-          invalidates.forEach((fsPath) => {
-            // worker should be new every time
-            moduleCache.delete(fsPath)
-            moduleCache.delete(`mock:${fsPath}`)
-          })
+          // worker should be new every time, invalidate its sub dependency
```

Comment:
Currently `invalidates` is not populated, and always contains only `fsPath`, so we can just remove it's usage entirely.

Reply:
In fact, every time vitest creates a new worker via the same URL, the `context` (or `self`) injected into the worker and its dependencies is not always up-to-date because of the `mod` cache.

So we need to clear both the worker module and its dependencies `mod` cache.

Reply:
For example:

```ts
// entry.test.ts
test('case1', async () => {
  new Worker(new URL('./worker', import.meta.url), {
    type: 'module'
  })
})

test('case2', async () => {
  new Worker(new URL('./worker', import.meta.url), {
    type: 'module'
  })
})
```

```ts
// worker.ts
import getDepSelf from './worker-dep'

console.log(getDepSelf() === self) // in case1, output is true while in case2, output is false
```

```ts
// worker-dep.ts
export default () => {
  return self
}
```

Reply:
Well, yes, I was hoping this is what this PR does, no?

Reply:
Yes, this PR does solve this problem

Reply:
Still, I don't think we need `invalidates` at all here, it is not used anywhere except to add `fsPath`. 

It's original idea was to keep track of dependencies and remove them after (what this PR does in a different way).

I was asking you to remove it

Reply:
So you mean it's OK to use `invalidateSubDepTree` to delete the dependency mod cache, besides that I should also remove the `invalidates` usage and not use it anymore？

Reply:
Yes

Reply:
Done. Sorry, I didn't catch what you mean before.